### PR TITLE
updated beta docs

### PR DESCRIPTION
**Pull request by v-uansari**

### CODE REVIEW

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md [hunk 1]
```diff
@@ -8,62 +19,103 @@ schema: 2.0.0
...
+KeyId               : aaaaaaaa-0b0b-1c1c-2d2d-333333333333
 StartDate           : 11/7/2016 12:00:00 AM
 Type                : Symmetric
 Usage               : Sign
 Value               : {49, 50, 51}
 ```
 
-The first command gets the ID of an application by using the Get-EntraBetaApplication (./Get-EntraBetaApplication.md)cmdlet.
-The command stores it in the $AppId variable.
+This example shows how to create an application key credential.
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:78]: In example section we need to add decription for Prarameters used in cmdlets  ObjectId,  CustomKeyIdentifier 
    StartDate .    Type,    Usage ,    Value

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md [hunk 2]
```diff
@@ -8,62 +19,103 @@ schema: 2.0.0
...
+    Value = $base64Value
+    StartDate = $cer.GetEffectiveDateString()
+    EndDate = $cer.GetExpirationDateString()
+}
+
+New-EntraBetaApplicationKeyCredential @params
+```
 
-The final command uses a certificate to add an application key credential.
+This example shows how to create an application key credential.
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:109]: In example section we need to add decription for Prarameters used in cmdlets ObjectId, CustomKeyIdentifier
StartDate . Type, Usage , Value,EndDate

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md [hunk 3]
```diff
@@ -41,6 +41,8 @@ The `New-EntraBetaApplicationPasswordCredential` cmdlet creates a password crede
 ### Example 1: Create a password credential
 
 ```powershell
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md:45]: In example section we need to add ObjectId description

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md [hunk 4]
```diff
@@ -55,23 +57,35 @@ This command creates new password credential for specified application.
...
 ```powershell
-New-EntraBetaApplicationPasswordCredential -ObjectId 'tttttttt-0000-2222-0000-aaaaaaaaaaaa' -CustomKeyIdentifier 'demoPassword'
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+$params = @{
+    ObjectId = 'tttttttt-0000-2222-0000-aaaaaaaaaaaa'
+    CustomKeyIdentifier = 'demoPassword'
+}
+
+New-EntraBetaApplicationPasswordCredential @params
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md:67]: In example description we need t to add used parameters description for ObjectId and CustomKeyIdentifier

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md [hunk 5]
```diff
@@ -55,23 +57,35 @@ This command creates new password credential for specified application.
...
 ```powershell
-New-EntraBetaApplicationPasswordCredential -ObjectId 'tttttttt-0000-2222-0000-aaaaaaaaaaaa' -StartDate (get-date).AddYears(0)
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+$params = @{
+    ObjectId = 'tttttttt-0000-2222-0000-aaaaaaaaaaaa'
+    StartDate = (Get-Date).AddYears(0)
+}
+
+New-EntraBetaApplicationPasswordCredential @params
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md:88]: In example description we need t to add used parameters description for ObjectId and StartDate

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md [hunk 6]
```diff
@@ -85,13 +99,20 @@ This command creates new password credential for specified application.
...
 ```powershell
-New-EntraBetaApplicationPasswordCredential -ObjectId 'tttttttt-0000-2222-0000-aaaaaaaaaaaa'-EndDate (get-date).AddYears(2)
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+$params = @{
+    ObjectId = 'tttttttt-0000-2222-0000-aaaaaaaaaaaa'
+    EndDate = (Get-Date).AddYears(2)
+}
+
+New-EntraBetaApplicationPasswordCredential @params
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationPasswordCredential.md:109]: In example description we need t to add used parameters description for ObjectId and EndDate

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationKeyCredential.md [hunk 7]
```diff
@@ -8,34 +18,50 @@ schema: 2.0.0
...
+```powershell
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+
+$params = @{
+    ObjectId = '33334444-dddd-5555-eeee-6666ffff7777'
+    KeyId = 'aaaaaaaa-0b0b-1c1c-2d2d-333333333333'
+}
+
+Remove-EntraBetaApplicationKeyCredential @params
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationKeyCredential.md:52]: In example description we need t to add used parameters description for ObjectId and KeyId

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md [hunk 8]
```diff
@@ -12,38 +22,39 @@ Removes a password credential from an application.
...
 
 ### Example 1: Remove an application password credential
-```
-PS C:\> $AppID = (Get-EntraBetaApplication -Top 1).ObjectId
-PS C:\> $KeyIDs = Get-EntraBetaApplicationPasswordCredential -ObjectId $AppId
-PS C:\> Remove-EntraBetaApplicationPasswordCredential -ObjectId $AppId -KeyId $KeyIds[0].KeyId
+```powershell
+$AppID = (Get-EntraApplication -Top 1).ObjectId
+$KeyIDs = Get-EntraBetaApplicationPasswordCredential -ObjectId $AppId
+Remove-EntraBetaApplicationPasswordCredential -ObjectId $AppId -KeyId $KeyIds[0].KeyId
```
Comment (v-varshamane) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md:41]: In example description we need t to add used parameters description for ObjectId and KeyId

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md [hunk 9]
```diff
@@ -9,42 +18,50 @@ schema: 2.0.0
...
 ## Examples
 
 ### Example 1: Remove an application password credential
-```
-PS C:\> $AppID = (Get-EntraBetaApplication -Top 1).ObjectId
-PS C:\> $KeyIDs = Get-EntraBetaApplicationPasswordCredential -ObjectId $AppId
-PS C:\> Remove-EntraBetaApplicationPasswordCredential -ObjectId $AppId -KeyId $KeyIds[0].KeyId
+
+```powershell
+$AppID = (Get-EntraApplication -Top 1).ObjectId
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md:42]:
```suggestion
$AppID = (Get-EntraBetaApplication -Top 1).ObjectId
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+## Inputs
+
+## Outputs
+
+## Notes
+
+## Related Links
+
+[New-EntraBetaApplicationPasswordCredential](New-EntraBetaApplicationPasswordCredential.md)
+[Remove-EntraBetaApplicationPasswordCredential](Remove-EntraBetaApplicationPasswordCredential.md)
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md:102]:
```suggestion

[Remove-EntraBetaApplicationPasswordCredential](Remove-EntraBetaApplicationPasswordCredential.md)
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md [hunk 11]
```diff
@@ -35,44 +46,102 @@ Get-EntraBetaDeletedApplication
 ```
 
 ## Description
-Retrieves the list of previously deleted applications
+
+Retrieves the list of previously deleted applications.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md:50]:
```suggestion
Retrieves the list of previously deleted applications.

Note: Deleted security groups are permanently removed and cannot be retrieved.
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md [hunk 12]
```diff
@@ -9,42 +18,50 @@ schema: 2.0.0
...
+$KeyIDs = Get-EntraBetaApplicationPasswordCredential -ObjectId $AppId
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md:43]: `Get-EntraBetaApplicationPasswordCredential` does not exist currently.

Comment (snehalkotwal) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md:43]: @SteveMutungi254 this Get-EntraBetaApplicationPasswordCredential cmdlet PR is not merged yet.

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationKeyCredential.md [hunk 13]
```diff
@@ -19,53 +18,47 @@ schema: 2.0.0
...
 
 This command gets the key credentials for the specified application.
+`-ObjectId` parameter specifies the ID of an application object in Microsoft Entra ID.
 
 ## Parameters
 
 ### -ObjectId
-Specifies a unique ID of an application in Microsoft Entra ID for which to get key credentials
+
+Specifies a unique ID of an application in Microsoft Entra ID for which to get key credentials.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationKeyCredential.md:58]:
```suggestion
Specifies a unique ID of an application in Microsoft Entra ID to retrieve key credentials. Use `Get-EntraBetaApplication` for more details.
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+The `Get-EntraBetaApplicationPasswordCredential` cmdlet receives the password credentials for a Microsoft Entra ID application. Specify `ObjectId` parameter to cmdlet receives the password credentials.
+
+## Examples
+
+### Example 1: Get password credential for specified application
+
+```powershell
+Connect-Entra -Scopes 'Application.Read.All'
+Get-EntraBetaApplicationPasswordCredential -ObjectId `aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb`
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md:42]:
```suggestion
Get-EntraBetaApplicationPasswordCredential -ObjectId 'aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb'
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+
+This command gets the password credential for specified application.
+
+- `-ObjectId` parameter specifies the ID of an application object in Microsoft Entra ID.
+
+## Parameters
+
+### -ObjectId
+
+The objectID of the application for which to get the password credential.
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md:59]:
```suggestion
The objectID of the application for which to get the password credential. Use `Get-EntraBetaApplication` for more details.
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md [hunk 16]
```diff
@@ -145,16 +219,19 @@ Accept wildcard characters: False
...
+
 System.Nullable\`1\[\[System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\]\] System.Nullable\`1\[\[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089\]\]
 
 ## Outputs
 
 ### System.Object
+
 ## Notes
 
 ## Related Links
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md:237]: Add related commands. Get-EntraBetaApplication is an obvious example.

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md [hunk 17]
```diff
@@ -35,44 +46,103 @@ Get-EntraBetaDeletedApplication
...
+```Output
+DisplayName Id                                   AppId                                SignInAudience PublisherDomain
+----------- --                                   -----                                -------------- ---------------
+TestApp1    aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb bbbbbbbb-1111-2222-3333-cccccccccccc AzureADMyOrg   contoso.com
 ```
 
-This example shows how an existing application was deleted and how the G-EntraBetaDeletedApplication cmdlet retrieves the application from the list of deleted applications
+This cmdlet retrieves deleted applications using SearchString parameter.  
+
+### Example 5: Get deleted applications filter by display name
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md:123]: Confirm this example works.

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md [hunk 18]
```diff
@@ -9,62 +19,118 @@ schema: 2.0.0
...
+
+```powershell
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+
+$AppId = (Get-EntraApplication -Top 1).Objectid
+$params = @{
+    ObjectId = $AppId
+    CustomKeyIdentifier = 'EntraPowerShellKey'
+    StartDate = '11/7/2016'
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:59]: Date format is: 2024-01-16T00:00:00Z

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md [hunk 19]
```diff
@@ -9,62 +19,118 @@ schema: 2.0.0
...
+
+The `New-EntraBetaApplicationKeyCredential` cmdlet creates a key credential for an application.
+
+An application can use this command along with `Remove-EntraBetaApplicationKeyCredential` to automate the rolling of its expiring keys.
+
+As part of the request validation, proof of possession of an existing key is verified before the action can be performed.
 
 ## Examples
 
 ### Example 1: Create a new application key credential
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:49]: Validate if this example works too.

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md [hunk 20]
```diff
@@ -9,62 +19,118 @@ schema: 2.0.0
...
+
+```powershell
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:53]:
```suggestion
Connect-Entra -Scopes 'Application.ReadWrite.All','Application.ReadWrite.OwnedBy'
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationKeyCredential.md [hunk 21]
```diff
@@ -9,34 +18,53 @@ schema: 2.0.0
...
-PS C:\> Remove-EntraBetaApplicationKeyCredential -ObjectId "3ddd22e7-a150-4bb3-b100-e410dea1cb84" -KeyId "6aa971c6-3040-45df-87ed-581c8c09ff2b"
+
+```powershell
+Connect-Entra -Scopes 'Application.ReadWrite.All' #Delegated Permission
+Connect-Entra -Scopes 'Application.ReadWrite.OwnedBy' #Application Permission
+
+$params = @{
+    ObjectId = '33334444-dddd-5555-eeee-6666ffff7777'
+    KeyId = 'aaaaaaaa-0b0b-1c1c-2d2d-333333333333'
+}
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationKeyCredential.md:50]:
```suggestion
Connect-Entra -Scopes 'Application.ReadWrite.All','Application.ReadWrite.OwnedBy'
$application = Get-EntraBetaApplication -Filter "displayName eq '<displayName>'"
$params = @{
    ObjectId = $application.Id
    KeyId = 'aaaaaaaa-0b0b-1c1c-2d2d-333333333333'
}
```

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md [hunk 22]
```diff
@@ -9,42 +18,50 @@ schema: 2.0.0
...
 
 ## Examples
 
 ### Example 1: Remove an application password credential
-```
-PS C:\> $AppID = (Get-EntraBetaApplication -Top 1).ObjectId
-PS C:\> $KeyIDs = Get-EntraBetaApplicationPasswordCredential -ObjectId $AppId
-PS C:\> Remove-EntraBetaApplicationPasswordCredential -ObjectId $AppId -KeyId $KeyIds[0].KeyId
+
+```powershell
```
Comment (SteveMutungi254) [inline module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Remove-EntraBetaApplicationPasswordCredential.md:41]: Add required permissions e.g. Connect-Entra -Scopes '....

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md (hunk 17 above)
Comment (v-pughosh) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md:123]: fixed the example

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md (hunk 16 above)
Comment (v-pughosh) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaDeletedApplication.md:237]: fixed

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md (hunk 19 above)
Comment (v-pughosh) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md:49]: fixed date

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md (hunk 18 above)
Comment (v-pughosh) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/New-EntraBetaApplicationKeyCredential.md]: updated

File: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md (hunk 14 above)
Comment (v-pughosh) [reply module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Get-EntraBetaApplicationPasswordCredential.md]: updated

**Review by SteveMutungi254 — APPROVED** [review]
Good to merge.

**Pull request merged**